tests/experimental/datasources/test_pandas_datasource.py

```diff
@@ -31,6 +52,133 @@ def csv_path() -> pathlib.Path:
     )
 
 
+class SpyInterrupt(RuntimeError):
+    """
+    Exception that may be raised to interrupt the control flow of the program
+    when a spy has already captured everything needed.
+    """
+
+    def __init__(self, message: str, captured=None) -> None:
+        self.captured = captured
+        super().__init__(message)
+
+
+@pytest.fixture
+def capture_reader_fn_params(monkeypatch: MonkeyPatch):
+    """
+    Capture the `reader_options` arguments being passed to the `PandasExecutionEngine`.
+
+    Note this fixture is heavily reliant on the implementation details of `PandasExecutionEngine`,
+    should this change this fixture will need to change.
+    """
+    captured_args: list[list] = []
+    captured_kwargs: list[dict[str, Any]] = []
+
+    def reader_fn_spy(*args, **kwargs):
+        logging.info(f"reader_fn_spy() called with...\n{args}\n{kwargs}")
+        captured_args.append(args)
+        captured_kwargs.append(kwargs)
+        raise SpyInterrupt(
+            "Reader options have been captured", {"args": args, "kwargs": kwargs}
+        )
+
+    monkeypatch.setattr(
+        great_expectations.execution_engine.pandas_execution_engine.PandasExecutionEngine,
+        "_get_reader_fn",
+        lambda *_: reader_fn_spy,
+        raising=True,
+    )
+
+    yield captured_args, captured_kwargs
```

Comment:
This is probably the most important part of these new tests.

It is currently only used by `test_data_asset_reader_options_passthrough`